pkg/cmd/root/help.go

```diff
@@ -0,0 +1,154 @@
+package root
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/OctopusDeploy/cli/pkg/constants/annotations"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/spf13/cobra"
+)
+
+func isRootCmd(cmd *cobra.Command) bool {
+	return cmd != nil && !cmd.HasParent()
+}
+
+func rootHelpFunc(cmd *cobra.Command, _ []string) {
+	namePadding := 12
+	coreCmds := []string{}
+	configurationCmds := []string{}
+	libraryCmds := []string{}
+	infrastructureCmds := []string{}
+	additionalCmds := []string{}
+	for _, c := range cmd.Commands() {
+		if c.Short == "" {
+			continue
+		}
+		if c.Hidden {
+			continue
+		}
+		s := rpad(c.Name()+":", namePadding) + c.Short
+		if _, ok := c.Annotations[annotations.IsCore]; ok {
+			coreCmds = append(coreCmds, s)
+		} else if _, ok := c.Annotations[annotations.IsConfiguration]; ok {
+			configurationCmds = append(configurationCmds, s)
+		} else if _, ok := c.Annotations[annotations.IsLibrary]; ok {
+			libraryCmds = append(libraryCmds, s)
+		} else if _, ok := c.Annotations[annotations.IsInfrastructure]; ok {
+			infrastructureCmds = append(infrastructureCmds, s)
+		} else {
+			additionalCmds = append(additionalCmds, s)
+		}
+	}
+
+	// Assume all additional cmds are core if no core cmds are found
+	if len(coreCmds) == 0 {
+		coreCmds = additionalCmds
+		additionalCmds = []string{}
```

Comment:
goland always complains at me for empty slice literals and wants me to use `make([]string, 0)` instead. I haven't looked into why.

I don't know if that's useful feedback, but 🤷